### PR TITLE
add skeletons into collection page

### LINKED ISSUE #988: Add skeletons to collections
When user opening collection, would be amazing to have a skeleton at 
- collection name
- creator
- analytics like items, owned, floor price volume traded 24h volume traded
- items 

https://kodadot.xyz/rmrk/collection/0E76E3AC15B4C1FA1E-FROST

image

### PR DESCRIPTION
**Pull request by fairywellos**
**Thank you for your contribution** to the KodaDot NFT gallery.
Let's do a quick check before the merge.
Your contribution will be an eternal part of the community codebase.

### PR type
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### Before submitting this PR, please make sure:
- [ ] Code builds clean without any errors or warnings
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've didn't break any original functionality
- [ ] I've posted screenshot of demonstrated change in this PR

### Optional
- [x] I've tested it on mobile and everything works
- [ ] I found edge cases

### What's new? (may be part of changelog)
- This PR closes #988 - <Add skeletons into CollectionItem for individual items>

### If issue had bounty label ($,$$,$$$)
- Fill up your KSM address: https://kodadot.xyz/transfer/?target=Dr1Ech2pQami4e6bBsZe6jTLSCuF4PWLW5DMbMfR3aPoRyF 

### Community participation
- [x] Are you at KodaDot Discord? If not yet, [let's join us - https://discord.gg/35hzy2dXXh](https://discord.gg/35hzy2dXXh)

### Screenshot
- In the initial loading
Screen Shot 2021-11-04 at 2 36 17 PM

### CODE REVIEW

**Comment by fairywellos** [comment]
@yangwao Created the PR with an initial changes and attached a screenshot as I thought that it's better to confirm how the skeletons looks like for each section (collection name, creator, analytics, and items).
Because of the delay of getting nft lists, the skeleton components disappears & we're still seeing blank screen. I'd love to confirm the behavior more in detail.

File: src/components/rmrk/Gallery/CollectionItem.vue [hunk 1]
```diff
@@ -13,7 +13,10 @@
             ></b-image>
           </div>
           <h1 class="title is-2">
-            {{ name }}
+            <template v-if="!nameLoading">
+              {{ name }}
+            </template>
+            <b-skeleton :active="nameLoading" :height="30" size="is-medium" :animated="true"></b-skeleton>
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/CollectionItem.vue:19]:
```suggestion
            <b-skeleton :active="nameLoading" size="is-large"></b-skeleton>
```
- animated property is by default true
- I would choose between using heigh or size but not both

File: src/components/rmrk/Gallery/CollectionItem.vue [hunk 2]
```diff
@@ -23,9 +26,10 @@
           <div class="label">
             {{ $t('creator') }}
           </div>
-          <div class="subtitle is-size-6">
+          <div v-if="issuer" class="subtitle is-size-6">
             <ProfileLink :address="issuer" :inline="true" :showTwitter="true"/>
           </div>
+          <b-skeleton :active="!issuer" width="20%" size="is-small" :animated="true"></b-skeleton>
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/CollectionItem.vue:32]:
```suggestion
          <b-skeleton :active="!issuer" width="20%" size="is-small"></b-skeleton>
```

File: src/components/rmrk/Gallery/CollectionItem.vue [hunk 3]
```diff
@@ -44,11 +48,13 @@
         </div>
       </div>
 
-      <CollectionActivity :nfts="stats" />
+      <CollectionActivity v-if="!statsLoading" :nfts="stats" />
+      <b-skeleton :active="statsLoading" height="30px" :animated="true"></b-skeleton>
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/CollectionItem.vue:52]: I don't know if this one is really necessary, but I would move the skeleton in CollectionActivity component

File: src/components/rmrk/Gallery/CollectionItem.vue [hunk 4]
```diff
@@ -44,11 +48,13 @@
...
 
       <div class="columns is-centered">
-        <div class="column is-8 has-text-centered">
+        <div class="column is-8 has-text-centered mt-5">
           <VueMarkdown :source="description" />
+          <b-skeleton :active="!description" height="80px" :animated="true"></b-skeleton>
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/CollectionItem.vue:57]:
```suggestion
          <b-skeleton :active="!description" height="80px"></b-skeleton>
```

File: src/components/rmrk/Gallery/CollectionItem.vue [hunk 5]
```diff
@@ -58,7 +64,21 @@
         </b-field>
       </Search>
 
-      <GalleryCardList :items="collection.nfts" />
+      <GalleryCardList :items="collection.nfts" />
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/CollectionItem.vue:67]:
```suggestion
      <GalleryCardList :items="collection.nfts" />
```

File: src/components/rmrk/Gallery/CollectionItem.vue [hunk 6]
```diff
@@ -58,7 +64,21 @@
...
+                    <div class="content">
+                        <p>
+                            <b-skeleton active></b-skeleton>
+                            <b-skeleton height="80px"></b-skeleton>
+                        </p>
+                    </div>
+                </div>
+            </article>
+        </section>
+    </template>
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/CollectionItem.vue:81]:
```suggestion

```
This should be removed as we already have skeleton for GalleryCard

File: src/components/rmrk/Gallery/CollectionItem.vue [hunk 7]
```diff
@@ -144,6 +165,11 @@ export default class CollectionItem extends Vue {
   }
 
   get name(): string {
+    if (this.collection.name === undefined) {
+      this.nameLoading = true
+    } else {
+      this.nameLoading = false
+    }
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/CollectionItem.vue:172]: What will happen if collection doesn't have name?

**Review by roiLeo — CHANGES_REQUESTED** [review]
- missing skeleton for collection image

**Comment by yangwao** [comment]
Bump @fairywellos are you able to finish this one?

**Comment by fairywellos** [comment]
> Bump @fairywellos are you able to finish this one?

Yes, I can start working from today.

File: src/components/rmrk/Gallery/CollectionItem.vue (hunk 1 above)
Comment (fairywellos) [reply src/components/rmrk/Gallery/CollectionItem.vue]: Pushed the fix

File: src/components/rmrk/Gallery/CollectionItem.vue (hunk 2 above)
Comment (fairywellos) [reply src/components/rmrk/Gallery/CollectionItem.vue]: pushed the fix

File: src/components/rmrk/Gallery/CollectionItem.vue (hunk 7 above)
Comment (roiLeo) [reply src/components/rmrk/Gallery/CollectionItem.vue:158]: something will not work here, if there is no collection name you will be displaying loading skeleton indefinitely

**Review by roiLeo — CHANGES_REQUESTED** [review]
I still don't see collection image skeleton.
You can use b-image placeholder cf. [Buefy doc](https://buefy.org/documentation/image/#placeholder)

**Comment by fairywellos** [comment]
pass issue to @roiLeo as I think that it's better to handle on his end with the necessary details further.

**Pull request closed without merging**